Apps/Sandcastle/gallery/Clustering.html

```diff
@@ -0,0 +1,172 @@
+            </td>
+        </tr>
+        <tr>
+            <td>minimum cluster size</td>
+            <td>
+                <input type="range" min="2" max="20" step="1" data-bind="value: minimumClusterSize, valueUpdate: 'input'">
+                <input type="text" size="2" data-bind="value: minimumClusterSize">
+            </td>
+        </tr>
+        </tbody>
+    </table>
+    <div id="toggleClustering"></div>
+    <div id="customStyling"></div>
+</div>
+<script id="cesium_sandcastle_script">
+function startup(Cesium) {
+    'use strict';
+//Sandcastle_Begin
+var viewer = new Cesium.Viewer('cesiumContainer');
+
+var options = {
+    camera : viewer.scene.camera,
+    canvas : viewer.scene.canvas
+};
+var dataSourcePromise = viewer.dataSources.add(Cesium.KmlDataSource.load('../../SampleData/kml/facilities/facilities.kml', options));
+dataSourcePromise.then(function(dataSource) {
+    var pixelRange = 15;
+    var minimumClusterSize = 3;
+
+    dataSource.clustering.enabled = true;
+    dataSource.clustering.pixelRange = pixelRange;
+    dataSource.clustering.minimumClusterSize = minimumClusterSize;
+
+    var viewModel = {
+        pixelRange: pixelRange,
+        minimumClusterSize: minimumClusterSize
+    };
+    Cesium.knockout.track(viewModel);
+
+    var toolbar = document.getElementById('toolbar');
+    Cesium.knockout.applyBindings(viewModel, toolbar);
+
+    function subscribeParameter(name) {
+        Cesium.knockout.getObservable(viewModel, name).subscribe(
+            function(newValue) {
+                dataSource.clustering[name] = newValue;
+            }
+        );
+    }
+
+    subscribeParameter('pixelRange');
+    subscribeParameter('minimumClusterSize');
+
+    Sandcastle.addToolbarButton('Toggle Clustering', function() {
+        dataSource.clustering.enabled = !dataSource.clustering.enabled;
+    }, 'toggleClustering');
+
+    var removeListener;
+
+    var pinBuilder = new Cesium.PinBuilder();
+    var pin50 = pinBuilder.fromText('50+', Cesium.Color.RED, 48).toDataURL();
+    var pin40 = pinBuilder.fromText('40+', Cesium.Color.ORANGE, 48).toDataURL();
+    var pin30 = pinBuilder.fromText('30+', Cesium.Color.YELLOW, 48).toDataURL();
+    var pin20 = pinBuilder.fromText('20+', Cesium.Color.GREEN, 48).toDataURL();
+    var pin10 = pinBuilder.fromText('10+', Cesium.Color.BLUE, 48).toDataURL();
+
+    var singleDigitPins = new Array(8);
+    for (var i = 0; i < singleDigitPins.length; ++i) {
+        singleDigitPins[i] = pinBuilder.fromText('' + (i + 2), Cesium.Color.VIOLET, 48).toDataURL();
+    }
+
+    function customStyle() {
+        if (Cesium.defined(removeListener)) {
+            removeListener();
+            removeListener = undefined;
+        } else {
+            removeListener = dataSource.clustering.clusterEvent.addEventListener(function(cluster, entity) {
+                entity.label = undefined;
+                entity.billboard = {};
```

Comment:
You probably want to set the origin to bottom here so the pins point is at the location of the cluster instead of at the center of the billboard.